Source/init.cpp

```diff
@@ -165,6 +164,27 @@ void init_archives()
 		}
 		LogVerbose("MPQ search paths:{}", message);
 	}
+}
+
+void init_devilutionx_archive()
+{
+	std::vector<std::string> paths;
+	get_mpq_search_paths(paths);
+
+	devilutionx_mpq = init_test_access(paths, "devilutionx.mpq");
+	HANDLE fh = nullptr;
+	if (!SFileOpenFile("font\\CharisSILB.ttf", &fh)) {
+		app_fatal("%s", _("Failed to load font resources.\n"
+		                  "\n"
+		                  "Make sure devilutionx.mpq is in the game folder and that it is up to date."));

```

Comment:
This message will be hard to read with no font loaded :)

Reply:
I used the following code as a model, assuming it would be similarly difficult to see the error message if the game window couldn't be created.
https://github.com/diasurgical/devilutionX/blob/792817048306a63f599be5efd7ffef1e68b93a47/Source/init.cpp#L233

Here is how it looks on Windows.
![image](https://user-images.githubusercontent.com/9203145/117564758-a58ed780-b07b-11eb-9fc2-39b6221be057.png)

As I mentioned above, 3DS simply crashes without displaying an error message. I am curious about what happens on Linux.

Reply:
Right that's how it looks if a window has not yet been created, guess this is a bit ealier in the process then it used to be.

Reply:
Yeah, this is the complication I was referring to on Discord. I had to load devilutionx.mpq early so the error message would be visible. If we can move away from TTF, this would no longer be necessary. That's why I think we should hold off on merging until we make a firm decision regarding TTF fonts.

Reply:
Well do we not have the same problem if using bitmap fonts? Since they would also need to be loaded from the file or diabdat.mpq? In fact we should probably adjust the dialog system to use the OS GUI when no font is currently loaded to avoid the "strange red empty dialog" issue.

Reply:
Have a small error case font converted to a byte array and embedded directly in the executable? :P

Reply:
Good point. I suppose you're right. And yeah, I guess loading early feels like a bit of a hack compared to using the OS GUI.

Reply:
I think OS GUI might only be available with SDL2 though so we should take that in to consideration as well.

Reply:
I believe this discussion is resolved with #1922.